state/applicationoffers.go

```diff
@@ -346,6 +339,24 @@ func (op *RemoveOfferOperation) internalRemove(offer *crossmodel.ApplicationOffe
 			})
 		}
 	}
+	// If the offer is being removed, then any proxies for applications on the
+	// consuming side also need to be removed.
+	remoteApps, err := op.offerStore.st.AllRemoteApplications()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	for _, remoteApp := range remoteApps {
+		if remoteApp.OfferUUID() != op.offer.OfferUUID {
```

Comment:
sad that we have to scan through all the remote apps in the model

Reply:
Yeah, there are only ever a few. I might add a query api